### PR TITLE
Add typings

### PR DESCRIPTION
**Pull request by OleksiiKachan**
- installed @types/node and typescript
- added tsconfig.json
- added types to package.json
- added types for metavm

### CODE REVIEW

**Comment by OleksiiKachan** [comment]
@tshemsedinov please review

File: package.json [hunk 1]
```diff
@@ -40,16 +41,19 @@
   },
   "scripts": {
     "lint": "eslint . && prettier --check \"**/*.js\" \"**/*.json\" \"**/*.md\" \"**/.*rc\" \"**/*.yml\"",
+    "types": "tsc -p types/tsconfig.json",
     "fmt": "prettier --write \"**/*.js\" \"**/*.json\" \"**/*.md\" \"**/.*rc\" \"**/*.yml\"",
-    "test": "eslint . && metatests test/"
+    "test": "npm run types && eslint . && metatests test/"
```
Comment (tshemsedinov) [inline package.json:46]:
```suggestion
    "test": "npm run lint && npm run types && metatests test/"
```

File: package.json [hunk 2]
```diff
@@ -40,16 +41,19 @@
   },
   "scripts": {
     "lint": "eslint . && prettier --check \"**/*.js\" \"**/*.json\" \"**/*.md\" \"**/.*rc\" \"**/*.yml\"",
```
Comment (tshemsedinov) [inline package.json:43]:
```suggestion
    "lint": "eslint . && prettier -c \"**/*.js\" \"**/*.json\" \"**/*.md\" \"**/.*rc\" \"**/*.ts\"",
```

File: package.json [hunk 3]
```diff
@@ -40,16 +41,19 @@
...
+    "types": "tsc -p types/tsconfig.json",
     "fmt": "prettier --write \"**/*.js\" \"**/*.json\" \"**/*.md\" \"**/.*rc\" \"**/*.yml\"",
```
Comment (tshemsedinov) [inline package.json:45]:
```suggestion
    "fmt": "prettier --write \"**/*.js\" \"**/*.json\" \"**/*.md\" \"**/.*rc\" \"**/*.ts\"",
```

File: types/tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  "compilerOptions": {
+    "target": "ESNext",
+    "moduleResolution": "node",
+    "strict": true,
+    "noEmit": true,
+    "baseUrl": ".",
+    "preserveWatchOutput": true
+  },
+  "include": ["*.d.ts"]
+}
```
Comment (tshemsedinov) [inline types/tsconfig.json:11]:
```suggestion
}

```

**Review by tshemsedinov — COMMENTED** [review]
Thanks for contribution, I'll fix some issues and land this today.

File: types/metavm.d.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  exports: any;
+}
+
+export function createScript(
+  name: string,
+  src: string,
+  options?: MetaScriptOptions
+): MetaScript;
+
+export function readScript(filePath: string, options?: BaseOptions): Promise<MetaScript>;
```
Comment (tshemsedinov) [inline types/metavm.d.ts:29]:
```suggestion
export function readScript(
  filePath: string,
  options?: BaseOptions
): Promise<MetaScript>;
```

File: package-lock.json [hunk 6]
```diff
@@ -1,8 +1,2378 @@
 {
   "name": "metavm",
   "version": "1.0.0",
-  "lockfileVersion": 1,
+  "lockfileVersion": 2,
```
Comment (tshemsedinov) [inline package-lock.json:4]:
```suggestion
  "lockfileVersion": 1,
```

Comment (tshemsedinov) [reply package-lock.json:4]: We still use npm 6.x

**Comment by tshemsedinov** [comment]
Landed in https://github.com/metarhia/metavm/commit/ccaf839e17c03965a5f0d304a0232d3b7d17b5a9 and https://github.com/metarhia/metavm/commit/491f99a9f40b9252deb698f5f10ba4ccd9e44776

**Pull request closed without merging**